osquery/core/query.cpp

```diff
@@ -122,6 +122,10 @@ Status Query::addNewResults(QueryDataTyped current_qd,
     saveQuery(name_, query_);
   }
 
+  // Counter should be zeroed. It will be reset in the case of a
+  // found update_db
+  counter = getQueryCounter(new_query);
```

Comment:
This fixed the tests but that kind of scares me. Either the tests are no good at testing what they intend to test or I don't fully understand the branching if logic. This line does nothing but sets the counter which is passed in.

Reply:
I agree, this looks wrong. We should deep dive into the test scenario and not set/reset the counter like this.